### PR TITLE
CRM-20022 Extended function to support multiple batch labels

### PR DESCRIPTION
**Pull request by Edzelopez**
----------------------------------------
* CRM-20022: Add Batch Title to Columns, Grouping tabs on Contribution Summary report
  https://issues.civicrm.org/jira/browse/CRM-20022

### CODE REVIEW

**Comment by Edzelopez** [comment]
@eileenmcnaughton Could you please review if this is a viable fix to prevent the issue that could arise with contributions in multiple batches?

File: CRM/Report/Form.php [hunk 1]
```diff
@@ -4909,7 +4909,12 @@ public function getLabels($options, $baoName, $fieldName) {
     $types = explode(',', $options);
     $labels = array();
     foreach ($types as $value) {
-      $labels[$value] = CRM_Core_PseudoConstant::getLabel($baoName, $fieldName, $value);
+      if ($labelValue = CRM_Core_PseudoConstant::getLabel($baoName, $fieldName, $value)) {
+        $labels[$value] = $labelValue;
+      }
+      else {
+        $labels[$value] = CRM_Core_DAO::getFieldValue($baoName, $value, $fieldName);
```
Comment (eileenmcnaughton) [inline CRM/Report/Form.php]: I think if you passed in the BAO Name for EntityBatch & added the pseudoconstant to the xml for batch_id in entity_batch you would not need to change this function like this, & I think adding metadata to the xml generally is a good way to do things (although you will need to regen the DAOs)

**Comment by eileenmcnaughton** [comment]
@Edzelopez I just realised... It might be better to filter the duplicates in mysql.... see the mysql man page for GROUP_CONCAT - you can use distinct with it

```
mysql> SELECT student_name,
    ->     GROUP_CONCAT(DISTINCT test_score
    ->               ORDER BY test_score DESC SEPARATOR ' ')
    ->     FROM student
    ->     GROUP BY student_name;
```

**Comment by Edzelopez** [comment]
@eileenmcnaughton This is very helpful. I think we should follow this approach and I will refactor the PR. Thank you!

**Comment by eileenmcnaughton** [comment]
@Edzelopez this one still needs another round doesn't it?

**Comment by Edzelopez** [comment]
@eileenmcnaughton I've made the changes now. Could you please review?

**Comment by eileenmcnaughton** [comment]
I'm happy with this - @monishdeb are you OK to do a final check per the others & merge assuming all good.

**Comment by monishdeb** [comment]
@eileenmcnaughton yup I have tested. Please merge it if the patch looks good to ya. Thanks

**Pull request merged**